assembl/tests/views_tests/test_graphql_vote_session.py

```diff
@@ -0,0 +1,254 @@
+    fetched_image_data = test_app.get(fetched_image['externalUrl']).body
+    assert fetched_image_data == new_image_data
+
+
+def test_graphql_update_vote_session(graphql_request, vote_session, test_app):
+    mutate_and_assert(graphql_request, vote_session.discussion_phase_id, test_app)
+
+
+simple_get_query = u"""
+    query VoteSesion($discussionPhaseId: Int!) {
+        voteSession(discussionPhaseId: $discussionPhaseId) {
+            id
+        }
+    }
+"""
+
+
+def vote_session_not_created(discussion_phase_id, graphql_request):
+    response = schema.execute(
+        simple_get_query,
+        context_value=graphql_request,
+        variable_values={"discussionPhaseId": discussion_phase_id}
+    )
+    return (response.errors is None) and (response.data['voteSession'] is None)
+
+
+def test_graphql_create_vote_session(graphql_request, timeline_vote_session, test_app):
+    assert vote_session_not_created(timeline_vote_session.id, graphql_request)
+    mutate_and_assert(graphql_request, timeline_vote_session.id, test_app)
+
+
+def test_graphql_create_vote_session_unauthenticated(graphql_unauthenticated_request, timeline_vote_session, test_app):
+    assert vote_session_not_created(timeline_vote_session.id, graphql_unauthenticated_request)
+    mutate_and_assert_fail(graphql_unauthenticated_request, timeline_vote_session.id)
+
+
+def mutate_and_assert_fail(graphql_request, discussion_phase_id):
+    new_title = u"updated vote session title"
+    response = schema.execute(
+        u"""
+            mutation UpdateVoteSession(
+              $discussionPhaseId: Int!
+              $headerImage: String
+              $titleEntries: [LangStringEntryInput]
+              $subTitleEntries: [LangStringEntryInput]
+              $instructionsSectionTitleEntries: [LangStringEntryInput]
+              $instructionsSectionContentEntries: [LangStringEntryInput]
+              $propositionsSectionTitleEntries: [LangStringEntryInput]
+            ) {
+              updateVoteSession(
+                discussionPhaseId: $discussionPhaseId
+                headerImage: $headerImage
+                titleEntries: $titleEntries
+                subTitleEntries: $subTitleEntries
+                instructionsSectionTitleEntries: $instructionsSectionTitleEntries
+                instructionsSectionContentEntries: $instructionsSectionContentEntries
+                propositionsSectionTitleEntries: $propositionsSectionTitleEntries
+              ) {
+                voteSession {
+                  id
+                }
+              }
+            }
+        """,
+        context_value=graphql_request,
+        variable_values={
+            "discussionPhaseId": discussion_phase_id,
+            "titleEntries": [{"localeCode": "en", "value": new_title}]
+        }
+    )
+
+    assert ("wrong credentials" in response.errors[0].message)
+
+
+def test_graphql_update_vote_session_unauthenticated(graphql_unauthenticated_request, vote_session):
```

Comment:
For permissions tests, you should add discussion_with_data fixture to have the permissions associated to the discussion.